src/schemas/Program/index.js

```diff
@@ -303,10 +316,35 @@ const resolvers = {
 	},
 	Query: {
 		program: async (obj, args, context, info) => {
+			const requestedFields = info.fieldNodes.flatMap((fieldNode) =>
+				fieldNode.selectionSet.selections.map((selection) => selection.name.value),
+			);
 			const { egoToken } = context;
```

Comment:
`requestedFields` returns an array of strings of the fields requested in the query, for example,` ['name', 'shortName', 'description']`. This is used to match to the private field variable above, `programServicePrivateFields` to determine whether the query is requesting private or public fields. And which of the resolvers to used, `resolveSingleProgram` (private GRPC) or `resolveHTTPProgram` (public HTTP).